### DISCUSSION TITLE
PeerTube for phpBB?

### DISCUSSION THREAD

**Discussion by Zepfanman** [s9e/TextFormatter#229] [Ideas]
Is this the right place to request PeerTube support? Would like to be able to embed [PeerTube](https://joinpeertube.org/) videos.

**Comment by JoshyPHP** [comment]
PeerTube is a software rather than a specific site, so you'd have to either add each instance individually or have a `peertube` media site with a dynamic host. At some point in the future I may add a helper for PeerTube as I do for Mastodon but either way that would require a user interface to be configured, so it doesn't help you.

Since you're asking about phpBB specifically, your best option would be to look into creating your own `.yml` file for the phpBB Media Embed PlugIn extension. I don't know exactly how it works but you can find some example on this page: https://www.phpbb.com/customise/db/extension/mediaembed/support/topic/174656

I'll leave this PHP implementation as an example.

```php
$configurator = new s9e\TextFormatter\Configurator;

$configurator->MediaEmbed->add(
	'diodezone',
	[
		'host'    => 'diode.zone',
		'extract' => '#/(?:videos/embed|w)/(?<id>[-\\w]+)#',
		'iframe'  => ['src' => 'https://diode.zone/videos/embed/{@id}']
	]
);

extract($configurator->finalize());

$text = 'https://diode.zone/w/df9591cb-7574-40ad-bdb5-eb06c555fb4e';
$xml  = $parser->parse($text);
$html = $renderer->render($xml);

die("$html\n");
```
```html
<span data-s9e-mediaembed="diodezone" style="display:inline-block;width:100%;max-width:640px"><span style="display:block;overflow:hidden;position:relative;padding-bottom:56.25%"><iframe allowfullscreen="" loading="lazy" scrolling="no" src="https://diode.zone/videos/embed/df9591cb-7574-40ad-bdb5-eb06c555fb4e" style="border:0;height:100%;left:0;position:absolute;width:100%"></iframe></span></span>
```
